In [ ]:
!mkdir bhx_dataset
%cd bhx_dataset

/content/bhx_dataset/bhx_dataset


In [ ]:
#Download the files of the dataset

!wget -r -N -c -np https://physionet.org/files/bhx-brain-bounding-box/1.1/


--2026-05-26 08:56:04--  https://physionet.org/files/bhx-brain-bounding-box/1.1/
Resolving physionet.org (physionet.org)... 18.13.52.205
Connecting to physionet.org (physionet.org)|18.13.52.205|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [text/html]
Saving to: ‘physionet.org/files/bhx-brain-bounding-box/1.1/index.html’

physionet.org/files     [ <=>                ]   1.01K  --.-KB/s    in 0s      

Last-modified header missing -- time-stamps turned off.
2026-05-26 08:56:04 (215 MB/s) - ‘physionet.org/files/bhx-brain-bounding-box/1.1/index.html’ saved [1030]

Loading robots.txt; please ignore errors.
--2026-05-26 08:56:04--  https://physionet.org/robots.txt
Reusing existing connection to physionet.org:443.
HTTP request sent, awaiting response... 200 OK
Length: 22 [text/plain]
Saving to: ‘physionet.org/robots.txt’

physionet.org/robot 100%[===================>]      22  --.-KB/s    in 0s      

2026-05-26 08:56:04 (12.2 MB/s) - ‘physionet.org/ro

# Brain Hemorrhage Extended (BHX): Bounding box extrapolation from thick to thin slice CT images

It was created to support:

- Brain hemorrhage detection
- Medical image annotation,
- Biomedical data analysis,
- Artificial intelligence research using CT scans.

The dataset contains:

CT imaging metadata,
hemorrhage annotations,
bounding box information,
manually labeled cases,
and extrapolated annotations.

All CT images with 3 slices or more.

CT Scan Images

        ↓

Radiologist Annotation

        ↓

Bounding Boxes

        ↓

Manual Label CSV

        ↓

Automatic Extrapolation

        ↓

Expanded Annotation Dataset

## Explore dataset structure

In [ ]:
!find . | head


.
./physionet.org
./physionet.org/files
./physionet.org/files/bhx-brain-bounding-box
./physionet.org/files/bhx-brain-bounding-box/1.1
./physionet.org/files/bhx-brain-bounding-box/1.1/2_Extrapolation_to_All_Series.csv
./physionet.org/files/bhx-brain-bounding-box/1.1/example_annotated_images.png
./physionet.org/files/bhx-brain-bounding-box/1.1/LICENSE.txt
./physionet.org/files/bhx-brain-bounding-box/1.1/SHA256SUMS.txt
./physionet.org/files/bhx-brain-bounding-box/1.1/index.html


###### The dataset contains CSV annotation tables, metadata files, and imaging-related resources. The CSV files are the main source of biomedical annotations used in this analysis.

In [ ]:
!find physionet.org/files/bhx-brain-bounding-box/1.1/ -type f | wc -l


7


## CSV files

In [ ]:
!wc -l physionet.org/files/bhx-brain-bounding-box/1.1/*.csv

    6284 physionet.org/files/bhx-brain-bounding-box/1.1/1_Initial_Manual_Labeling.csv
   39669 physionet.org/files/bhx-brain-bounding-box/1.1/2_Extrapolation_to_All_Series.csv
   27204 physionet.org/files/bhx-brain-bounding-box/1.1/3_Extrapolation_to_Selected_Series.csv
   73157 total


#### The extrapolated datasets contain substantially more annotations than the manually labeled dataset

### Manual Labeling csv

Contains:

-Manually created hemorrhage annotations.

These annotations were generated by medical experts.

### Contains Expert Annotations


In [ ]:
!head  -n 10 physionet.org/files/bhx-brain-bounding-box/1.1/1_Initial_Manual_Labeling.csv

,SOPInstanceUID,SeriesInstanceUID,StudyInstanceUID,data,labelName,labelType
95,1.2.276.0.7230010.3.1.4.296485376.1.1521713089.1721949,1.2.276.0.7230010.3.1.3.296485376.1.1521713089.1721932,1.2.276.0.7230010.3.1.2.296485376.1.1521713088.1721651,"{'x': 236.16402999999997, 'y': 151.01577, 'width': 50.876979999999996, 'height': 62.99052}",Intraventricular,handDrawn
96,1.2.276.0.7230010.3.1.4.296485376.1.1521713089.1721953,1.2.276.0.7230010.3.1.3.296485376.1.1521713089.1721932,1.2.276.0.7230010.3.1.2.296485376.1.1521713088.1721651,"{'x': 232.93374, 'y': 156.66877, 'width': 57.33755, 'height': 48.45427}",Intraventricular,handDrawn
97,1.2.276.0.7230010.3.1.4.296485376.1.1521713089.1721959,1.2.276.0.7230010.3.1.3.296485376.1.1521713089.1721932,1.2.276.0.7230010.3.1.2.296485376.1.1521713088.1721651,"{'x': 230.51106000000001, 'y': 147.78548999999998, 'width': 57.33752, 'height': 54.10725}",Intraventricular,handDrawn
98,1.2.276.0.7230010.3.1.4.296485376.1.1521713089.1721973,1.2.276.0.7230010.3.1.

In [ ]:
!wc -l physionet.org/files/bhx-brain-bounding-box/1.1/1_Initial_Manual_Labeling.csv

6284 physionet.org/files/bhx-brain-bounding-box/1.1/1_Initial_Manual_Labeling.csv


#### The dataset contains 6284 rows, representing manually annotated biomedical records related to brain hemorrhage findings in CT imaging studies.

#### Since the first line corresponds to the CSV header, the dataset includes 6283 manually annotated bounding boxes, matching the description provided in the original BHX publication.

In [ ]:
!cut -d',' -f2 \
physionet.org/files/bhx-brain-bounding-box/1.1/1_Initial_Manual_Labeling.csv \
| sort | uniq | wc -l
# numer of images /  each SOPInstanceUID ,one CT image slice

3559


### The dataset contains approximately 3558 unique CT image slices, plus the header, confirming the statistics described in the BHX publication. Multiple hemorrhage annotations may exist within the same image, explaining why the number of bounding boxes exceeds the number of images.

## Search in Manual Labelling csv


In [ ]:
!grep -i "hemorrhage" physionet.org/files/bhx-brain-bounding-box/1.1/1_Initial_Manual_Labeling.csv | head
#  labels

In [ ]:
!grep -i "subdural" physionet.org/files/bhx-brain-bounding-box/1.1/1_Initial_Manual_Labeling.csv | head

3631,1.2.276.0.7230010.3.1.4.296485376.1.1521714221.2013535,1.2.276.0.7230010.3.1.3.296485376.1.1521714221.2013494,1.2.276.0.7230010.3.1.2.296485376.1.1521714218.2012943,"{'x': 268.76867000000004, 'y': 111.75635, 'width': 103.5881, 'height': 227.22552000000002}",Subdural,handDrawn
3632,1.2.276.0.7230010.3.1.4.296485376.1.1521714221.2013533,1.2.276.0.7230010.3.1.3.296485376.1.1521714221.2013494,1.2.276.0.7230010.3.1.2.296485376.1.1521714218.2012943,"{'x': 109.85932, 'y': 123.26613, 'width': 96.16244, 'height': 120.29586}",Subdural,handDrawn
3633,1.2.276.0.7230010.3.1.4.296485376.1.1521714221.2013543,1.2.276.0.7230010.3.1.3.296485376.1.1521714221.2013494,1.2.276.0.7230010.3.1.2.296485376.1.1521714218.2012943,"{'x': 120.62653999999999, 'y': 155.5678, 'width': 34.52937, 'height': 90.22189}",Subdural,handDrawn
3634,1.2.276.0.7230010.3.1.4.296485376.1.1521714221.2013509,1.2.276.0.7230010.3.1.3.296485376.1.1521714221.2013494,1.2.276.0.7230010.3.1.2.296485376.1.1521714218.2012943,"{'x': 142.90

#### The extracted rows correspond to subdural hemorrhage annotations. These records can be used to study the frequency and distribution of this hemorrhage subtype.

In [ ]:
!grep -i "subdural" physionet.org/files/bhx-brain-bounding-box/1.1/1_Initial_Manual_Labeling.csv | wc -l

1849


### Different types of acute hemorrhage

In [ ]:
!grep -i "Subarachnoid" physionet.org/files/bhx-brain-bounding-box/1.1/1_Initial_Manual_Labeling.csv | head

180,1.2.276.0.7230010.3.1.4.296485376.1.1521713681.1874071,1.2.276.0.7230010.3.1.3.296485376.1.1521713681.1873998,1.2.276.0.7230010.3.1.2.296485376.1.1521713681.1873994,"{'x': 181.90804, 'y': 176.92525, 'width': 108.32158000000001, 'height': 92.4344}",Subarachnoid,handDrawn
181,1.2.276.0.7230010.3.1.4.296485376.1.1521713681.1874027,1.2.276.0.7230010.3.1.3.296485376.1.1521713681.1873998,1.2.276.0.7230010.3.1.2.296485376.1.1521713681.1873994,"{'x': 176.85302, 'y': 269.35966, 'width': 67.88153, 'height': 102.54443}",Subarachnoid,handDrawn
182,1.2.276.0.7230010.3.1.4.296485376.1.1521713681.1874029,1.2.276.0.7230010.3.1.3.296485376.1.1521713681.1873998,1.2.276.0.7230010.3.1.2.296485376.1.1521713681.1873994,"{'x': 171.07588, 'y': 253.4725, 'width': 82.3244, 'height': 130.70803999999998}",Subarachnoid,handDrawn
183,1.2.276.0.7230010.3.1.4.296485376.1.1521713681.1873999,1.2.276.0.7230010.3.1.3.296485376.1.1521713681.1873998,1.2.276.0.7230010.3.1.2.296485376.1.1521713681.1873994,"{'x': 159.5215

In [ ]:
!grep -i "Subarachnoid" physionet.org/files/bhx-brain-bounding-box/1.1/1_Initial_Manual_Labeling.csv | wc -l

1746


In [ ]:
!grep -E "Subdural|Epidural|Intraventricular" \
physionet.org/files/bhx-brain-bounding-box/1.1/1_Initial_Manual_Labeling.csv | head -n 10


95,1.2.276.0.7230010.3.1.4.296485376.1.1521713089.1721949,1.2.276.0.7230010.3.1.3.296485376.1.1521713089.1721932,1.2.276.0.7230010.3.1.2.296485376.1.1521713088.1721651,"{'x': 236.16402999999997, 'y': 151.01577, 'width': 50.876979999999996, 'height': 62.99052}",Intraventricular,handDrawn
96,1.2.276.0.7230010.3.1.4.296485376.1.1521713089.1721953,1.2.276.0.7230010.3.1.3.296485376.1.1521713089.1721932,1.2.276.0.7230010.3.1.2.296485376.1.1521713088.1721651,"{'x': 232.93374, 'y': 156.66877, 'width': 57.33755, 'height': 48.45427}",Intraventricular,handDrawn
97,1.2.276.0.7230010.3.1.4.296485376.1.1521713089.1721959,1.2.276.0.7230010.3.1.3.296485376.1.1521713089.1721932,1.2.276.0.7230010.3.1.2.296485376.1.1521713088.1721651,"{'x': 230.51106000000001, 'y': 147.78548999999998, 'width': 57.33752, 'height': 54.10725}",Intraventricular,handDrawn
98,1.2.276.0.7230010.3.1.4.296485376.1.1521713089.1721973,1.2.276.0.7230010.3.1.3.296485376.1.1521713089.1721932,1.2.276.0.7230010.3.1.2.296485376.1.1521713

In [ ]:
!cut -d',' -f9 physionet.org/files/bhx-brain-bounding-box/1.1/1_Initial_Manual_Labeling.csv | sort | uniq -c | sort -nr | head


   1849 Subdural
   1746 Subarachnoid
   1324 Intraparenchymal
    740 Chronic
    558 Intraventricular
     66 Epidural
      1 


### Extrapolation to all series csv

Contains:

automatically extrapolated annotations.


Annotations were computationally propagated to additional slices.

In [ ]:
!head -n 10 /content/bhx_dataset/physionet.org/files/bhx-brain-bounding-box/1.1/2_Extrapolation_to_All_Series.csv

,SOPInstanceUID,SeriesInstanceUID,StudyInstanceUID,data,labelName,labelType
0,1.2.276.0.7230010.3.1.4.296485376.1.1521713091.1722412,1.2.276.0.7230010.3.1.3.296485376.1.1521713090.1722057,1.2.276.0.7230010.3.1.2.296485376.1.1521713088.1721651,"{'x': 320.95899, 'y': 235.81072999999998, 'width': 30.68771, 'height': 52.4921}",Intraventricular,selectedThinSlices
1,1.2.276.0.7230010.3.1.4.296485376.1.1521713091.1722416,1.2.276.0.7230010.3.1.3.296485376.1.1521713090.1722057,1.2.276.0.7230010.3.1.2.296485376.1.1521713088.1721651,"{'x': 320.95899, 'y': 235.81072999999998, 'width': 30.68771, 'height': 52.4921}",Intraventricular,selectedThinSlices
2,1.2.276.0.7230010.3.1.4.296485376.1.1521713091.1722372,1.2.276.0.7230010.3.1.3.296485376.1.1521713090.1722057,1.2.276.0.7230010.3.1.2.296485376.1.1521713088.1721651,"{'x': 320.95899, 'y': 235.81072999999998, 'width': 30.68771, 'height': 52.4921}",Intraventricular,selectedThinSlices
3,1.2.276.0.7230010.3.1.4.296485376.1.1521713091.1722384,1.2.276.0.72

In [ ]:
!wc -l /content/bhx_dataset/physionet.org/files/bhx-brain-bounding-box/1.1/2_Extrapolation_to_All_Series.csv

39669 /content/bhx_dataset/physionet.org/files/bhx-brain-bounding-box/1.1/2_Extrapolation_to_All_Series.csv


### Extrapolation to selected series csv

Contains:

extrapolated annotations for selected CT series only.

A curated subset,
filtered data,
or quality-controlled extrapolation results.

In [ ]:
!head -n 10 /content/bhx_dataset/physionet.org/files/bhx-brain-bounding-box/1.1/3_Extrapolation_to_Selected_Series.csv

,SOPInstanceUID,SeriesInstanceUID,StudyInstanceUID,data,labelName,labelType
0,1.2.276.0.7230010.3.1.4.296485376.1.1521713091.1722412,1.2.276.0.7230010.3.1.3.296485376.1.1521713090.1722057,1.2.276.0.7230010.3.1.2.296485376.1.1521713088.1721651,"{'x': 320.95899, 'y': 235.81072999999998, 'width': 30.68771, 'height': 52.4921}",Intraventricular,selectedThinSlices
1,1.2.276.0.7230010.3.1.4.296485376.1.1521713091.1722416,1.2.276.0.7230010.3.1.3.296485376.1.1521713090.1722057,1.2.276.0.7230010.3.1.2.296485376.1.1521713088.1721651,"{'x': 320.95899, 'y': 235.81072999999998, 'width': 30.68771, 'height': 52.4921}",Intraventricular,selectedThinSlices
2,1.2.276.0.7230010.3.1.4.296485376.1.1521713091.1722372,1.2.276.0.7230010.3.1.3.296485376.1.1521713090.1722057,1.2.276.0.7230010.3.1.2.296485376.1.1521713088.1721651,"{'x': 320.95899, 'y': 235.81072999999998, 'width': 30.68771, 'height': 52.4921}",Intraventricular,selectedThinSlices
3,1.2.276.0.7230010.3.1.4.296485376.1.1521713091.1722384,1.2.276.0.72

In [ ]:
!wc -l /content/bhx_dataset/physionet.org/files/bhx-brain-bounding-box/1.1/3_Extrapolation_to_Selected_Series.csv

27204 /content/bhx_dataset/physionet.org/files/bhx-brain-bounding-box/1.1/3_Extrapolation_to_Selected_Series.csv


In [ ]:
!cut -d',' -f9 physionet.org/files/bhx-brain-bounding-box/1.1/3_Extrapolation_to_Selected_Series.csv | sort | uniq -c | sort -nr | head


   7904 Subdural
   7596 Subarachnoid
   5264 Intraparenchymal
   3672 Chronic
   2219 Intraventricular
    548 Epidural
      1 


## Subdural hemorrhage cases

In [ ]:
!grep -i "subdural" physionet.org/files/bhx-brain-bounding-box/1.1/1_Initial_Manual_Labeling.csv > subdural_cases.csv

In [ ]:
!head subdural_cases.csv

3631,1.2.276.0.7230010.3.1.4.296485376.1.1521714221.2013535,1.2.276.0.7230010.3.1.3.296485376.1.1521714221.2013494,1.2.276.0.7230010.3.1.2.296485376.1.1521714218.2012943,"{'x': 268.76867000000004, 'y': 111.75635, 'width': 103.5881, 'height': 227.22552000000002}",Subdural,handDrawn
3632,1.2.276.0.7230010.3.1.4.296485376.1.1521714221.2013533,1.2.276.0.7230010.3.1.3.296485376.1.1521714221.2013494,1.2.276.0.7230010.3.1.2.296485376.1.1521714218.2012943,"{'x': 109.85932, 'y': 123.26613, 'width': 96.16244, 'height': 120.29586}",Subdural,handDrawn
3633,1.2.276.0.7230010.3.1.4.296485376.1.1521714221.2013543,1.2.276.0.7230010.3.1.3.296485376.1.1521714221.2013494,1.2.276.0.7230010.3.1.2.296485376.1.1521714218.2012943,"{'x': 120.62653999999999, 'y': 155.5678, 'width': 34.52937, 'height': 90.22189}",Subdural,handDrawn
3634,1.2.276.0.7230010.3.1.4.296485376.1.1521714221.2013509,1.2.276.0.7230010.3.1.3.296485376.1.1521714221.2013494,1.2.276.0.7230010.3.1.2.296485376.1.1521714218.2012943,"{'x': 142.90

In [ ]:
!echo "BHX BIOMEDICAL DATASET REPORT" > report.txt

!echo "=============================" >> report.txt

!echo "TOTAL FILES:" >> report.txt
!find . -type f | wc -l >> report.txt

!echo "TOTAL RECORDS:" >> report.txt
!wc -l physionet.org/files/bhx-brain-bounding-box/1.1/1_Initial_Manual_Labeling.csv >> report.txt

!echo "HEMORRHAGE LABEL DISTRIBUTION:" >> report.txt
!cut -d',' -f2 physionet.org/files/bhx-brain-bounding-box/1.1/1_Initial_Manual_Labeling.csv | sort | uniq -c | sort -nr >> report.txt



In [ ]:
!cat report.txt | head -n 20

BHX BIOMEDICAL DATASET REPORT
TOTAL FILES:
10
TOTAL RECORDS:
6284 physionet.org/files/bhx-brain-bounding-box/1.1/1_Initial_Manual_Labeling.csv
HEMORRHAGE LABEL DISTRIBUTION:
      8 1.2.276.0.7230010.3.1.4.296485376.1.1521713835.1915205
      8 1.2.276.0.7230010.3.1.4.296485376.1.1521713835.1915201
      8 1.2.276.0.7230010.3.1.4.296485376.1.1521713835.1915183
      8 1.2.276.0.7230010.3.1.4.296485376.1.1521713835.1915145
      7 1.2.276.0.7230010.3.1.4.296485376.1.1521714388.2049357
      7 1.2.276.0.7230010.3.1.4.296485376.1.1521714388.2049321
      7 1.2.276.0.7230010.3.1.4.296485376.1.1521713835.1915217
      7 1.2.276.0.7230010.3.1.4.296485376.1.1521713835.1915211
      7 1.2.276.0.7230010.3.1.4.296485376.1.1521713835.1915199
      7 1.2.276.0.7230010.3.1.4.296485376.1.1521713835.1915179
      7 1.2.276.0.7230010.3.1.4.296485376.1.1521713835.1915141
      7 1.2.276.0.7230010.3.1.4.296485376.1.1521713835.1915139
      7 1.2.276.0.7230010.3.1.4.296485376.1.1521712979.1693575



### Conclusion

Using shell scripting tools such as grep, cut, sort,
uniq, wc, and regular expressions, the BHX dataset
was explored and processed to identify hemorrhage
annotations, quantify biomedical imaging records,
and extract clinically relevant information from
structured CT metadata.

The analysis demonstrates how shell scripting can
support biomedical data mining and annotation analysis
in medical imaging datasets.
